web/src/main/java/org/springframework/security/web/firewall/ClientErrorRequestRejectedHandler.java

```diff
@@ -0,0 +1,17 @@
+package org.springframework.security.web.firewall;
+
+import java.io.IOException;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * A simple implementation of {@link RequestRejectedHandler} that sends and error with status code 400.
+ */
+public class ClientErrorRequestRejectedHandler implements RequestRejectedHandler {
+	@Override
+	public void handle(HttpServletRequest request, HttpServletResponse response,
+			RequestRejectedException requestRejectedException) throws IOException {
+		response.sendError(HttpServletResponse.SC_BAD_REQUEST);
```

Comment:
I think we should add the exception message to sendError so that the user knows what went wrong.

Reply:
Are we sure that the message does not leak any security relevant information?

Reply:
I'm adding logging here, should it be `WARN`, `INFO`, or `DEBUG`?
```java
if (logger.isDebugEnabled()) {
	logger.debug("Rejecting request due to: "+requestRejectedException.getMessage(),
			requestRejectedException);
}
```

Reply:
Debug please.